frontend/components/search-results.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Link } from "react-router-dom";
+import PropTypes from 'prop-types';
+import { BadgeExamples } from './badge-examples';
+import badgeExampleData from '../../badge-examples.json';
+import { prepareExamples, predicateFromQuery } from '../lib/prepare-examples';
+import { baseUri, longCache } from '../constants';
+
+
+export default class SearchResults extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.props = props;
+  }
+
+  static propTypes = {
+    category: PropTypes.string,
+    query: PropTypes.string,
+    clickHandler: PropTypes.func.isRequired,
+  }
+
+  prepareExamples(category) {
+    const examples = category ? badgeExampleData.filter(example => example.category.id === category) : badgeExampleData;
+    return prepareExamples(examples, () => predicateFromQuery(this.props.query));
+  }
+
+  renderExamples() {
+    return (
+      <BadgeExamples
+        categories={this.preparedExamples}
+        onClick={this.props.clickHandler}
+        baseUri={baseUri}
+        longCache={longCache} />
+    );
+  }
+
+  renderCategories() {
```

Comment:
Is this always rendering empty categories? That seems fine, but maybe calling it `renderEmptyCategories` would make that more obvious.

Reply:
How about `renderCategoryHeadings()`? That seems like quite a good description of what we are doing here.

Reply:
Works for me.